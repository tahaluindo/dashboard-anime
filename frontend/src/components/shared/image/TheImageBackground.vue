<template>
  <div class="absolute inset-0 z-0">
    <the-picture v-if="srcset" :srcset="srcset" :alt="alt" />
    <img
      class="object-cover min-w-full min-h-full"
      v-if="src"
      :src="src"
      :alt="alt"
    />
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from "vue";
import { ISrcset } from "./model/ISrcset";
import ThePicture from "./ThePicture.vue";

export default defineComponent({
  components: { ThePicture },
  props: {
    srcset: {
      type: Object as PropType<ISrcset>,
    },
    src: {
      type: String,
    },
    alt: {
      type: String,
      required: true,
    },
  },
});
</script>

<style lang="postcss"></style>
