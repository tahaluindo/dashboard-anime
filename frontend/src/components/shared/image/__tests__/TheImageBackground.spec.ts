import { mount } from "@vue/test-utils";
import TheImageBackground from "../TheImageBackground.vue";
import { srcsetMock } from "../__mocks__/srcset.mock";

describe("TheImageBackground test suit", () => {
  it("renders the component with src prop", () => {
    const wrapper = mount(TheImageBackground, {
      props: {
        src: "https:/localhost/test",
        alt: "image",
      },
      shallow: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the component with src prop", () => {
    const srcset = srcsetMock;

    const wrapper = mount(TheImageBackground, {
      props: {
        srcset,
        alt: "image",
      },
      shallow: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
