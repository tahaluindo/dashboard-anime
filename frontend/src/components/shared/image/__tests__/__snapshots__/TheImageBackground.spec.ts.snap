// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheImageBackground test suit renders the component with src prop 1`] = `
"<div class=\\"absolute inset-0 z-0\\">
  <!--v-if--><img class=\\"object-cover min-w-full min-h-full\\" src=\\"https:/localhost/test\\" alt=\\"image\\">
</div>"
`;

exports[`TheImageBackground test suit renders the component with src prop 2`] = `
"<div class=\\"absolute inset-0 z-0\\">
  <the-picture-stub srcset=\\"[object Object]\\" alt=\\"image\\"></the-picture-stub>
  <!--v-if-->
</div>"
`;
